utils/wasm-builder/src/checks.rs

```diff
@@ -0,0 +1,48 @@
+use gear_wasm_instrument::parity_wasm::elements::{Internal, Module, Type};
+
+#[derive(Debug)]
+pub enum Error {
+    ExportSectionMissing,
+    ImportSectionMissing,
+    FunctionSectionMissing,
+    TypeSectionMissing,
+    CodeSectionMissing,
+    UnexpectedFunctionExport(String),
+    MissingEntryFunction,
+    InvalidExportFunctionSignature,
+}
+
+pub fn do_checks(module: Module) -> Result<(), Error> {
+
+    let exports = module.export_section().ok_or(Error::ExportSectionMissing)?;
+    let imports = module.import_section().ok_or(Error::ImportSectionMissing)?;
+    let functions = module.function_section().ok_or(Error::FunctionSectionMissing)?;
+    let types = module.type_section().ok_or(Error::TypeSectionMissing)?;
+    let code = module.code_section().ok_or(Error::CodeSectionMissing)?;
+
+    let imported_functions = imports.functions();
+
+    // While this is checked in Code::new, we might as well check it when building wasm program
```

Comment:
This should have already been checked by wasm-builder. Your recent changes in `get_exports` are also checked.
https://github.com/gear-tech/gear/blob/c2733ecc57ddad27ce7c23a72789258276f89c2f/utils/wasm-builder/src/optimize.rs#L132-L134

Reply:
Should I perhaps do this in Code::try_new fully instead, that way it gets checked there instead of in wasm-builder